### PR TITLE
Componentising and refactoring

### PR DESCRIPTION
**Pull request by Reinachan**
Componentising the search results

Styled them more.

### CODE REVIEW

File: codegen.yml [hunk 1]
```diff
@@ -15,3 +15,9 @@ generates:
       baseTypesPath: types/graphql.ts
     plugins:
       - typescript-operations
+
+  src/possibleTypes.json:
```
Comment (toyhammered) [inline codegen.yml:19]: Can we move this to be in the same directory as `graphql.ts` (types/*)

File: src/components/media/SearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+// Components
+import Title from '../../styles/components/Title';
+import { SubtypeTag } from '../../styles/components/Tag';
+import Search from 'src/styles/layouts/Search';
+import Poster from 'src/styles/components/Poster';
+
+interface Props {
+  data: SearchMediaByTitleQuery;
+}
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:19]: we don't need an interface for this.

File: src/components/media/SearchResults.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+export default function SearchResults({ data }: Props): ReactElement {
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:21]: Can we arrow function this?

File: src/components/media/SearchResults.tsx (hunk 2 above)
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:19]: we don't need an interface for this.

File: src/components/media/SearchResults.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                key={media?.id}
+                className="media-link"
+                to={`/${media?.type}/${media?.id}`}>
+                <div className="media">
+                  <div className="media-inside">
+                    <Poster
+                      className="poster-image"
+                      style={{
+                        backgroundImage:
+                          'url(' + media?.posterImage.original.url + ')',
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:38]:
```suggestion
                          'url(' + media?.posterImage?.original.url + ')',
```

File: src/components/media/SearchResults.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                      }}
+                    />
+                    <Title className="media-title">
+                      {media?.titles.canonical}
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:42]:
```suggestion
                      {media?.titles?.canonical}
```

File: src/components/media/SearchResults.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                    </Title>
+
+                    <SubtypeTag className="subtype-tag">
+                      {media?.type}
+                    </SubtypeTag>
+                    <div className="season-date">
+                      {seasonYear(media?.startDate)}
+                    </div>
+                    <div className="search-description">
+                      <span>{media?.description.en}</span>
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:52]:
```suggestion
                      <span>{media?.description?.en}</span>
```

File: src/components/media/SearchResults.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                    </div>
+                  </div>
+                  <div className="media-fade-out"></div>
+                </div>
+              </Link>
+              <div className="on-kitsu">
+                <a
+                  href={`https://kitsu.io/${media?.type.toLowerCase()}/${
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:60]:
```suggestion
                  href={`https://kitsu.io/${media?.type?.toLowerCase()}/${
```

File: src/components/media/search_media.graphql [hunk 8]
```diff
@@ -5,19 +5,24 @@ query SearchMediaByTitle(
...
-        original {
-          url
-        }
-      }
+      ...media
+    }
+  }
+}
+
+fragment media on Media {
```
Comment (toyhammered) [inline src/components/media/search_media.graphql:13]: I think fragment names need to be unique across the app, should make this more descriptive like `mediaSearchFields` or something.

**Review by toyhammered — CHANGES_REQUESTED** [review]
few things to change, but looks great!

File: codegen.yml (hunk 1 above)
Comment (Reinachan) [reply codegen.yml:19]: Yeah, totally. I just wanted to get it done really. Will fix tomorrow

File: src/components/media/SearchResults.tsx (hunk 3 above)
Comment (Reinachan) [reply src/components/media/SearchResults.tsx:21]: Oh, right, I was using a snippet for setting it up and it uses classic functions by default. I still personally prefer them for functional components as it makes it much clearer where the boundaries are, but we can use arrow functions if you prefer that

File: src/components/media/SearchResults.tsx (hunk 2 above)
Comment (Reinachan) [reply src/components/media/SearchResults.tsx:19]: Same as above, I was using a snippet and wasn't really thinking

File: src/components/media/search_media.graphql (hunk 8 above)
Comment (Reinachan) [reply src/components/media/search_media.graphql:13]: Sure. No problem. Will fix tomorrow.

File: src/components/media/SearchResults.tsx (hunk 2 above)
Comment (Reinachan) [reply src/components/media/SearchResults.tsx:18]: I couldn't remove the interface for that for some reason. It just threw errors at me everywhere.

File: src/components/media/search_media.graphql (hunk 8 above)
Comment (Reinachan) [reply src/components/media/search_media.graphql]: Fixed

File: src/components/media/SearchResults.tsx (hunk 3 above)
Comment (Reinachan) [reply src/components/media/SearchResults.tsx]: Fixed

File: codegen.yml (hunk 1 above)
Comment (Reinachan) [reply codegen.yml]: Fixed

File: package.json [hunk 9]
```diff
@@ -2,21 +2,25 @@
...
   "version": "0.1.0",
   "dependencies": {
-    "@apollo/client": "^3.2.7",
+    "@apollo/client": "^3.3.0",
     "@fortawesome/fontawesome-free": "^5.15.1",
     "@fortawesome/fontawesome-svg-core": "^1.2.32",
     "@fortawesome/free-solid-svg-icons": "^5.15.1",
     "@fortawesome/react-fontawesome": "^0.1.14",
     "@types/node": "^14.14.20",
+    "@types/react-lazyload": "^3.0.0",
```
Comment (toyhammered) [inline package.json:11]: please add to dev dependencies. Nothing typescript should be here.

File: src/components/media/SearchResults.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+import Title from '../../styles/components/media/Title';
+import { SubtypeTag } from '../../styles/components/ui/Tag';
+import Search from 'src/styles/layouts/Search';
+import PosterImage from 'src/components/media/PosterImage';
+
+interface Props {
+  data: SearchMediaByTitleQuery;
+}
+
+const SearchResults = ({ data }: Props): ReactElement => {
```
Comment (toyhammered) [inline src/components/media/SearchResults.tsx:20]:
```suggestion
const SearchResults = ({ data }: { data: SearchMediaByTitleQuery}): ReactElement => {
```

File: src/components/media/SearchResults.tsx (hunk 2 above)
Comment (toyhammered) [reply src/components/media/SearchResults.tsx:18]: left an example below as a suggest change

File: src/routes/Home/SearchMedia.tsx [hunk 11]
```diff
@@ -107,48 +110,21 @@ const SearchMedia = (): ReactElement => {
...
-          ) : (
-            <div></div>
-          )}
-        </div>
+        {loading ? (
+          <Loading></Loading>
+        ) : error ? (
+          <span className="search-error">error</span>
+        ) : !searchIsRendered ? (
+          <></>
```
Comment (toyhammered) [inline src/routes/Home/SearchMedia.tsx:118]: why do we do this?

File: src/routes/Home/searchMedia.graphql [hunk 12]
```diff
@@ -0,0 +1,29 @@
+query SearchMediaByTitle(
+  $first: Int!
+  $title: String!
+  $media_type: MediaTypeEnum
+) {
+  searchMediaByTitle(first: $first, title: $title, mediaType: $media_type) {
+    nodes {
+      ...media
```
Comment (toyhammered) [inline src/routes/Home/searchMedia.graphql:8]:
```suggestion
      ...mediaSearchFields
```

File: src/app.tsx [hunk 13]
```diff
@@ -11,7 +11,7 @@ import {
 import { ApolloProvider } from '@apollo/client';
 import client from './client';
 
-import Media from './components/media';
+import Media from './routes/Home/index';
```
Comment (toyhammered) [inline src/app.tsx:14]: Why was this changed?

**Review by toyhammered — CHANGES_REQUESTED** [review]
Looking good! Just a few more corrections. We might need to change some of the directories where stuff is stored but for now it is fine (some of the naming seems strange).

File: package.json (hunk 9 above)
Comment (Reinachan) [reply package.json:11]: Ah, I'm not all that familiar with the difference between dev dependencies or not yet, will do

File: src/routes/Home/SearchMedia.tsx (hunk 11 above)
Comment (Reinachan) [reply src/routes/Home/SearchMedia.tsx:118]: To remove the search results if you remove your search query entirely and leave the field blank. Better UX imo.

File: src/app.tsx (hunk 13 above)
Comment (Reinachan) [reply src/app.tsx:14]: Because I moved around files in one of my commits. Restructured things into something that makes more sense.

File: src/components/media/SearchResults.tsx (hunk 10 above)
Comment (Reinachan) [reply src/components/media/SearchResults.tsx:20]: We decided on keeping this as is

**Review by toyhammered — APPROVED** [review]
LGTM! Just repoint your branch to main.
